### ISSUE TITLE
selected content parent tag remove

### ISSUE THREAD

**Issue by prabha-912** [artf/grapesjs#2101]
https://github.com/artf/grapesjs/issues/906#issuecomment-381403858

I need to remove the html content from editor.getSelected(). Now I'm using, 
```
var selectedContent = editor.getSelected();
selectedContent.set('content', '');
selectedContent.components(MYHTML);
```

While I use **selectedContent.set('content', '');** it remove all the inner html. But didn't remove the parent tag. For example, 
Before: `<h2 id="it4a"><p>Sample Update Check Done</p></h2>`
After: `<h2 id="it4a"></h2>`

After set the components, its like `<h2 id="it4a">NEW HTML</h2>`. How to remove the parent tags while set the innerHtml empty? 

Guide me how to solve this?

**Comment by afdallah** [comment]
Have you seen this [#1077 ](https://github.com/artf/grapesjs/issues/1077#issuecomment-386444814) answer by @artf
I think it's suited with your case.

```javascript
const component = editor.getSelected();
const coll = component.collection;
const at = coll.indexOf(component);
coll.remove(component);
return coll.add('<div>New content</div>', { at });
```

**Comment by prabha-912** [comment]
Yeah I tried @afdallah, It returns indexOf undefined. Let me try again. Thanks for your response :)

**Comment by prabha-912** [comment]
Hi @afdallah @artf I tried the above concept. It works fine in first time. While I try to update second time, it didn't work. 
![indexOf](https://user-images.githubusercontent.com/45986663/60251831-e7c72980-98e6-11e9-945d-f5f536c04507.png)

**Comment by prabha-912** [comment]
![collection](https://user-images.githubusercontent.com/45986663/60252394-c74b9f00-98e7-11e9-907f-cbbc6c636d82.png)
It remove the collection from the selected content. While I update the content, it needs to re-create right? (by coll.add()). but it didn't. How to solve this?

**Comment by afdallah** [comment]
> ![collection](https://user-images.githubusercontent.com/45986663/60252394-c74b9f00-98e7-11e9-907f-cbbc6c636d82.png)
> It remove the collection from the selected content. While I update the content, it needs to re-create right? (by coll.add()). but it didn't. How to solve this?

It would be easier for us to help if you provide a simple live demo. Jsfiddle / jsbin / repl

**Comment by prabha-912** [comment]
I couldn't create live demo. Now I updated the data by refreshing the collection. But stuck with another issue. While I update, the css didn't return fully from modal.( I'm using this to get data, `document.getElementById('gjs-cm-css').innerText`) If the css content is small, it returns correctly. If the css content is too large, then scroll comes. At that time, visible css only returned by the element.  
For example, 
![edit-1](https://user-images.githubusercontent.com/45986663/60499210-a616f400-9cd5-11e9-899d-8fc33f38a7b8.png)
in this screenshot, document.getElementById returns ~30 lines of css. But really I have the following amount of css in modal. ```#ikey4{
  box-sizing:border-box;
  display:table;
  width:100%;
  max-width:100%;
  padding-top:10px;
  padding-right:10px;
  padding-bottom:10px;
  padding-left:10px;
}
#is70l{
  box-sizing:border-box;
  display:table-cell;
  min-height:75px;
  width:70%;
  float:left;
}
#ilmfc{
  box-sizing:border-box;
  width:auto;
  margin-top:10px;
  margin-right:10px;
  margin-bottom:10px;
  margin-left:10px;
}
#iahq4{
  box-sizing:border-box;
  width:100%;
  text-align:justify;
  padding-top:10px;
  padding-right:10px;
  padding-bottom:10px;
  padding-left:10px;
  margin-top:0px;
  margin-right:0px;
  margin-bottom:0px;
  margin-left:0px;
}
#i4pda{
  box-sizing:border-box;
  font-family:-webkit-standard;
  text-align:start;
  line-height:22px;
}
#ijvdg{
  box-sizing:border-box;
  display:table;
  width:100%;
  max-width:100%;
  padding-top:10px;
  padding-right:10px;
  padding-bottom:10px;
  padding-left:10px;
}
#iiv0d{
  box-sizing:border-box;
  width:100%;
  text-align:left;
  font-weight:400;
  padding-top:20px;
  padding-right:20px;
  padding-bottom:20px;
  padding-left:20px;
  margin-top:0px;
  margin-right:0px;
  margin-bottom:0px;
  margin-left:0px;
}
#iigvd{
  box-sizing:border-box;
  display:table;
  width:100%;
  max-width:100%;
  padding-top:10px;
  padding-right:10px;
  padding-bottom:10px;
  padding-left:10px;
}
#ik4gv{
  box-sizing:border-box;
  width:100%;
  text-align:left;
  padding-top:20px;
  padding-right:20px;
  padding-bottom:20px;
  padding-left:20px;
  margin-top:0px;
  margin-right:0px;
  margin-bottom:0px;
  margin-left:0px;
}
#ibsp4{
  box-sizing:border-box;
  display:table;
  width:100%;
  min-width:40%;
  max-width:100%;
  padding-top:10px;
  padding-right:10px;
  padding-bottom:10px;
  padding-left:10px;
}
#ieqvw{
  box-sizing:border-box;
  display:table-cell;
  min-height:75px;
  width:33.333333%;
  float:left;
}
#i36ix{
  box-sizing:border-box;
  text-align:center;
  width:auto;
  padding-top:20px;
  padding-right:20px;
  padding-bottom:20px;
  padding-left:20px;
  margin-top:10px;
  margin-right:10px;
  margin-bottom:10px;
  margin-left:10px;
}
#iansh{
  box-sizing:border-box;
  display:table-cell;
  min-height:75px;
  width:33.333333%;
  float:left;
}
#iikan{
  box-sizing:border-box;
  text-align:center;
  width:auto;
  padding-top:20px;
  padding-right:20px;
  padding-bottom:20px;
  padding-left:20px;
  margin-top:10px;
  margin-right:10px;
  margin-bottom:10px;
  margin-left:10px;
}
#ijibe{
  box-sizing:border-box;
  display:table-cell;
  min-height:75px;
  width:33.333333%;
  float:left;
}
#iy5or{
  box-sizing:border-box;
  text-align:center;
  width:auto;
  padding-top:20px;
  padding-right:20px;
  padding-bottom:20px;
  padding-left:20px;
  margin-top:10px;
  margin-right:10px;
  margin-bottom:10px;
  margin-left:10px;
}
#i1f9j{
  box-sizing:border-box;
  display:table;
  width:100%;
  max-width:100%;
  padding-top:10px;
  padding-right:10px;
  padding-bottom:10px;
  padding-left:10px;
}
#iys0t{
  box-sizing:border-box;
  display:table-cell;
  min-height:75px;
  width:30%;
  float:left;
}
#in8i6{
  box-sizing:border-box;
  width:auto;
  margin-top:10px;
  margin-right:10px;
  margin-bottom:10px;
  margin-left:10px;
}
#ie3js{
  box-sizing:border-box;
  width:auto;
  height:auto;
}
#im784{
  box-sizing:border-box;
  height:32%;
  width:64%;
  display:block;
  max-width:100%;
  min-height:auto;
  margin-left:auto;
  margin-right:auto;
}
#iolr2{
  box-sizing:border-box;
  display:table;
  width:100%;
  max-width:100%;
  padding-top:10px;
  padding-right:10px;
  padding-bottom:10px;
  padding-left:10px;
}
.c4970{
  box-sizing:border-box;
  display:table;
  width:100%;
  max-width:100%;
  padding-top:10px;
  padding-right:10px;
  padding-bottom:10px;
  padding-left:10px;
}
.c1186{
  box-sizing:border-box;
  text-align:center;
  width:100%;
  padding-top:20px;
  padding-right:20px;
  padding-bottom:20px;
  padding-left:20px;
  margin-top:0px;
  margin-right:0px;
  margin-bottom:0px;
  margin-left:0px;
}
``` Is there any way to get full css from it?

**Comment by prabha-912** [comment]
I got the CSS  too. By using `this.htmlEditor..getContent()`, `this.cssEditor..getContent()` I get all the content properly. :). Thanks for your response @afdallah.
